pytorch_lightning/trainer/trainer.py

```diff
@@ -1746,6 +1754,7 @@ def __setup_profiler(self) -> None:
         self.profiler.setup(stage=self.state.fn._setup_fn, local_rank=local_rank, log_dir=self.log_dir)
 
     def _log_device_info(self) -> None:
+        # TODO(justusschock): should this include the nvidia gpus only or mps as well?
```

Comment:
IMO MPS as well

Just need to change `torch.cuda.is_available` for `GPUAccelerator.is_available()`

Where `GPUAccelerator` is something like:

```python
if MPSAccelerator.is_available():
    GPUAccelerator = MPSAccelerator
else:
    GPUAccelerator = CUDAccelerator
```

And perhaps include a "type: mps | cuda" in the message

Reply:
currently this is separate, but can easily change this :)